crates/bevy_ecs/src/component.rs

```diff
@@ -336,57 +408,68 @@ impl ComponentDescriptor {
         }
     }
 
-    fn new_non_send<T: Any>(storage_type: StorageType) -> Self {
+    /// Create a new `ComponentDescriptor` for a `!Send` resource.
+    ///
+    /// The [`StorageType`] for resources is always [`TableStorage`].
+    fn non_send_from_type<T: Any>() -> Self {
         Self {
             name: Cow::Borrowed(std::any::type_name::<T>()),
-            storage_type,
+            storage_type: StorageType::Table,
             is_send_and_sync: false,
             type_id: Some(TypeId::of::<T>()),
             layout: Layout::new::<T>(),
             drop: needs_drop::<T>().then_some(Self::drop_ptr::<T> as _),
         }
     }
-
-    #[inline]
-    pub fn storage_type(&self) -> StorageType {
-        self.storage_type
-    }
-
-    #[inline]
-    pub fn type_id(&self) -> Option<TypeId> {
-        self.type_id
-    }
-
-    #[inline]
-    pub fn name(&self) -> &str {
-        self.name.as_ref()
-    }
 }
 
-#[derive(Debug, Default)]
+/// The metadata store for all components within a [`World`].
+///
+/// A public read-only API is accessbile. To initialize a new component,
+/// use [`World::init_component`], [`World::init_resource`], [`World::init_non_send_resource`] or
+/// [`World::init_component_with_descriptor`].  Initialized components cannot be removed or
```

Comment:
❓ What is the difference between an initialized vs uninitialized component?

Reply:
IIRC components have to be assigned a `ComponentId` on a per-world basis, which occurs during initialization.

Would be good to expland on here though.